xmtp_mls/src/storage/encrypted_store/native.rs

```diff
@@ -145,7 +155,11 @@ impl NativeDb {
             pool.state().connections
         );
 
-        Ok(pool.get()?)
+        // Turn of writitng by default
+        let mut conn = pool.get()?;
+        conn.batch_execute("PRAGMA query_only = ON;")?;
```

Comment:
this will reset once the connection is out of scope, so you need to put it in the customizer `on_acquire`. the writeconn is OK as long as its not dropped/held on the NativeDb for the duration of execution